src/main/java/me/magnet/consultant/Consultant.java

```diff
@@ -395,8 +413,11 @@ public void registerService(int port) {
 			String serviceId = id.getInstance().get();
 			String serviceName = id.getServiceName();
 			String serviceHost = id.getHostName().orElse(InetAddress.getLoopbackAddress().getHostAddress());
-			Check check = new Check("http://" + serviceHost + ":" + port + "/" + healthEndpoint, HEALTH_CHECK_INTERVAL);
-			ServiceRegistration registration = new ServiceRegistration(serviceId, serviceName, serviceHost, port, check);
+			Collection<String> tags = id.getTags();
+			Check check = new Check("http://" + serviceHost + ":" + port + "/" + healthEndpoint,
+					HEALTH_CHECK_INTERVAL);
+			ServiceRegistration registration = new ServiceRegistration(serviceId, serviceName, serviceHost, port,
+					check, tags.toArray(new String[0]));
```

Comment:
The `new String[0]` just leads to a useless allocation. Do `new String[tags.size()]` instead.